### ISSUE TITLE
Target branch filter for Pull Requests

### ISSUE THREAD

**Issue by tpiekarski** [go-gitea/gitea#26308]
### Feature Description

The possibility to filter pull requests by target branch would be a great feature for gitea (see mockup).

### Screenshots
![image](https://github.com/go-gitea/gitea/assets/5697640/60baa56d-334a-4a60-8367-d8135d3429ae)

**Comment by puni9869** [comment]
You can use the label.

**Comment by wolfogre** [comment]
I don't think it's redundant. Let's take Gitea itself as an example, sometimes I want to filter out PRs which are not backports.

I could do that by:

<img width="1252" alt="image" src="https://github.com/go-gitea/gitea/assets/9418365/93519c42-a2cf-4b88-976d-b09c9d6718d3">

Or 

<img width="1291" alt="image" src="https://github.com/go-gitea/gitea/assets/9418365/6859995b-deea-43bd-9dae-2bacc9db4b0a">

**Comment by lunny** [comment]
More and more filters, we really need a search syntax like GH's

**Comment by puni9869** [comment]
Agree with @lunny

**Comment by tpiekarski** [comment]
> You can use the label. It will be redundant to have a branch filter.

No. How should I use labels? How is such filter redundant? Should I label every PR and the label contain the target branch name? 

Now that's redundant. Every PR has the source and target branch already. 

Well, look around. Github and gitlab got such filter, so I think they're useful 😊

If you want I can take a closer look and draft a PR.

**Comment by puni9869** [comment]
Pr is always welcome. But before that I would like the see the how you will implement. A small work flow or approach before you start.
Currently
1. Github is having query parser.
2. Gitlab is having enrich tag based query.

You have my full support on this feature.
